pkg/language/checks.go

```diff
@@ -390,6 +392,165 @@ func (r *Reader) checkPost(method *concepts.Method) {
 	}
 }
 
+func (r *Reader) checkSearch(method *concepts.Method) {
+	// Get the input and output parameters:
+	var in []*concepts.Parameter
+	var out []*concepts.Parameter
+	for _, parameter := range method.Parameters() {
+		if parameter.In() {
+			in = append(in, parameter)
+		}
+		if parameter.Out() {
+			out = append(out, parameter)
+		}
+	}
+
+	// Get the reference to the and to the version:
+	resource := method.Owner()
+	version := resource.Owner()
+
+	for _, parameter := range out {
+		if !parameter.Type().IsScalar() && !parameter.Type().IsList() {
+			r.reporter.Errorf(
+				"Parameter '%s' should be scalar or list but it is %s",
+				parameter, parameter.Type().Kind(),
+			)
+		}
+	}
+
+	// Only struct parameters:
+	for _, parameter := range in {
+		if !parameter.Type().IsScalar() && !parameter.Type().IsStruct() {
+			r.reporter.Errorf(
+				"Parameter '%s' should be scalar or struct but it is '%s'",
+				parameter, parameter.Type().Kind(),
+			)
+		}
+	}
+
+	// Exactly one list parameter:
```

Comment:
from this point onward this function is identical to `checkList()`.
Can you DRY them a bit?
One option is extract helpers `checkPageParam`, `checkSizeParam` etc ...

Reply:
Done